I'm getting different results in CI and locally:

* in CI:
```py
$ mypy --version && mypy
mypy 1.20.0 (compiled: yes)
tests/unit/bokeh/document/test_callbacks__document.py: note: In member "test_unhold" of class "TestDocumentCallbackManager":
tests/unit/bokeh/document/test_callbacks__document.py:394:9: error: Statement
is unreachable  [unreachable]
            cm.unhold()
            ^~~~~~~~~~~
Found 1 error in 1 file (checked 670 source files)
```
* locally:
```py
$ mypy --version && mypy
mypy 1.20.0 (compiled: yes)
src/bokeh/sphinxext/_internal/bokeh_directive.py:30: note: In module imported here:
/home/mateusz/conda/envs/bk_py3.14/lib/python3.14/site-packages/sphinx/util/docutils.py:53:57: error: Type statement is only
supported in Python 3.12 and greater  [syntax]
        type _DocutilsSettings = docutils.frontend.Values  # pyright: ignore[reportDeprecated]  # ty: ignore[deprecated]
                                                            ^
Found 1 error in 1 file (errors prevented further checking)
```

I can get the `unreachable` error if I run `mypy tests/`.

--- a/.github/workflows/bokeh-ci-full.yml
+++ b/.github/workflows/bokeh-ci-full.yml
@@ -70,9 +70,39 @@ jobs:
       - name: Run codebase checks
         run: pytest --color=yes tests/codebase
 
-      - name: MyPy
+  typing:
+    needs: build
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-24.04, macos-latest, windows-latest]
+        python-version: ['3.11', '3.12', '3.13', '3.14']
+
+    steps:
+      - uses: actions/checkout@v6
+
+      - name: Prepare Environment
+        uses: ./.github/workflows/composite/test-setup
+        with:
+          test-env: ${{ matrix.python-version }}
+          source-tree: 'keep'
+
+      - name: Ensure Python version
+        run: if [[ ! "$(python --version | cut -d' ' -f2)" == "${{ matrix.python-version }}"* ]]; then exit 1; fi
+
+      - name: List installed software
+        uses: ./.github/workflows/composite/list-software
+
+      - name: Check types with mypy
+        run: |
+          mypy --version && mypy
+
+      - name: Check types in examples with mypy
         if: success() || failure()
-        run: mypy --version && mypy
+        run: |
+          mypy --version && mypy --config-file examples/mypy.cfg examples/basic/scatters/color_scatter.py
 
   examples:
     needs: build
@@ -257,7 +287,7 @@ jobs:
           path: docs/bokeh/docs-html.tgz
 
   notify-nightly-failure:
-    needs: [build, codebase, examples, unit-test, minimal-deps, core-deps, documentation]
+    needs: [build, codebase, typing, examples, unit-test, minimal-deps, core-deps, documentation]
     if: failure() && github.event_name == 'schedule'
     runs-on: ubuntu-latest
 
@@ -299,7 +329,7 @@ jobs:
           *Automated notification from [Bokeh-CI-Full](${{ github.server_url }}/${{ github.repository }}/actions/workflows/bokeh-ci-full.yml)*"
 
   notify-pr:
-    needs: [build, codebase, examples, unit-test, minimal-deps, core-deps, documentation]
+    needs: [build, codebase, typing, examples, unit-test, minimal-deps, core-deps, documentation]
     if: always() && github.event_name == 'workflow_dispatch'
     runs-on: ubuntu-latest
 

--- a/.github/workflows/bokeh-ci.yml
+++ b/.github/workflows/bokeh-ci.yml
@@ -64,9 +64,39 @@ jobs:
       - name: Run codebase checks
         run: pytest --color=yes tests/codebase
 
-      - name: MyPy
+  typing:
+    needs: build
+    runs-on: ${{ matrix.os }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-24.04]
+        python-version: ['3.11', '3.14']
+
+    steps:
+      - uses: actions/checkout@v6
+
+      - name: Prepare Environment
+        uses: ./.github/workflows/composite/test-setup
+        with:
+          test-env: ${{ matrix.python-version }}
+          source-tree: 'keep'
+
+      - name: Ensure Python version
+        run: if [[ ! "$(python --version | cut -d' ' -f2)" == "${{ matrix.python-version }}"* ]]; then exit 1; fi
+
+      - name: List installed software
+        uses: ./.github/workflows/composite/list-software
+
+      - name: Check types with mypy
+        run: |
+          mypy --version && mypy
+
+      - name: Check types in examples with mypy
         if: success() || failure()
-        run: mypy --version && mypy
+        run: |
+          mypy --version && mypy --config-file examples/mypy.cfg examples/basic/scatters/color_scatter.py
 
   examples:
     needs: build

--- a/conda/environment-test-3.11.yml
+++ b/conda/environment-test-3.11.yml
@@ -66,7 +66,7 @@ dependencies:
   # pip dependencies
   - pip
   - pip:
-    - mypy >=1.19
+    - mypy >=1.20
     # docs
     - bokeh_sampledata
     - pydata_sphinx_theme

--- a/conda/environment-test-3.12.yml
+++ b/conda/environment-test-3.12.yml
@@ -66,7 +66,7 @@ dependencies:
   # pip dependencies
   - pip
   - pip:
-    - mypy >=1.19
+    - mypy >=1.20
     # docs
     - bokeh_sampledata
     - pydata_sphinx_theme

--- a/conda/environment-test-3.13.yml
+++ b/conda/environment-test-3.13.yml
@@ -66,7 +66,7 @@ dependencies:
   # pip dependencies
   - pip
   - pip:
-    - mypy >=1.19
+    - mypy >=1.20
     # docs
     - bokeh_sampledata
     - pydata_sphinx_theme

--- a/conda/environment-test-3.14.yml
+++ b/conda/environment-test-3.14.yml
@@ -66,7 +66,7 @@ dependencies:
   # pip dependencies
   - pip
   - pip:
-    - mypy >=1.19
+    - mypy >=1.20
     # docs
     - bokeh_sampledata
     - pydata_sphinx_theme

--- a/conda/environment-test-minimal-deps.yml
+++ b/conda/environment-test-minimal-deps.yml
@@ -54,7 +54,7 @@ dependencies:
   # pip dependencies
   - pip
   - pip:
-    - mypy >=1.11
+    - mypy >=1.20
     # docs
     # omit bokeh_sampledata
     - pydata_sphinx_theme

--- a/examples/mypy.cfg
+++ b/examples/mypy.cfg
@@ -1,0 +1,3 @@
+[mypy]
+python_version = 3.11
+strict = true

--- a/tests/unit/bokeh/document/test_callbacks__document.py
+++ b/tests/unit/bokeh/document/test_callbacks__document.py
@@ -391,7 +391,7 @@ class TestDocumentCallbackManager:
 
         cm.hold(policy)
         assert cm.hold_value == policy
-        cm.unhold()
+        cm.unhold() # type: ignore[unreachable]
         assert cm.hold_value is None
 
     @patch("bokeh.document.callbacks.DocumentCallbackManager.trigger_on_change")

